tests/ci/ci_definitions.py

```diff
@@ -0,0 +1,781 @@
+    batches: Optional[List[int]] = None
+    # Do not set it. A list of batches to await. It will be set in runtime in accordance with ci cache and ci settings
+    pending_batches: Optional[List[int]] = None
+
+    def with_properties(self, **kwargs: Any) -> "JobConfig":
+        res = copy.deepcopy(self)
+        for k, v in kwargs.items():
+            assert hasattr(self, k), f"Setting invalid attribute [{k}]"
+            setattr(res, k, v)
+        return res
+
+    def get_required_build(self) -> str:
+        assert self.required_builds
+        return self.required_builds[0]
+
+
+class CommonJobConfigs:
+    """
+    Common job configs
+    """
+
+    BUILD_REPORT = JobConfig(
+        job_name_keyword="build_check",
+        run_command="build_report_check.py",
+        digest=DigestConfig(
+            include_paths=[
+                "./tests/ci/build_report_check.py",
+                "./tests/ci/upload_result_helper.py",
+            ],
+        ),
+        runner_type=Runners.STYLE_CHECKER_ARM,
+    )
+    COMPATIBILITY_TEST = JobConfig(
+        job_name_keyword="compatibility",
+        digest=DigestConfig(
+            include_paths=["./tests/ci/compatibility_check.py"],
+            docker=["clickhouse/test-old-ubuntu", "clickhouse/test-old-centos"],
+        ),
+        run_command="compatibility_check.py",
+        runner_type=Runners.STYLE_CHECKER,
+    )
+    INSTALL_TEST = JobConfig(
+        job_name_keyword="install",
+        digest=DigestConfig(
+            include_paths=["./tests/ci/install_check.py"],
+            docker=["clickhouse/install-deb-test", "clickhouse/install-rpm-test"],
+        ),
+        run_command='install_check.py "$CHECK_NAME"',
+        runner_type=Runners.STYLE_CHECKER,
+        timeout=900,
+    )
+    STATELESS_TEST = JobConfig(
+        job_name_keyword="stateless",
+        digest=DigestConfig(
+            include_paths=[
+                "./tests/ci/functional_test_check.py",
+                "./tests/queries/0_stateless/",
+                "./tests/clickhouse-test",
+                "./tests/config",
+                "./tests/*.txt",
+            ],
+            exclude_files=[".md"],
+            docker=["clickhouse/stateless-test"],
+        ),
+        run_command='functional_test_check.py "$CHECK_NAME"',
+        runner_type=Runners.FUNC_TESTER,
+        timeout=10800,
+    )
+    STATEFUL_TEST = JobConfig(
+        job_name_keyword="stateful",
+        digest=DigestConfig(
+            include_paths=[
+                "./tests/ci/functional_test_check.py",
+                "./tests/queries/1_stateful/",
+                "./tests/clickhouse-test",
+                "./tests/config",
+                "./tests/*.txt",
+            ],
+            exclude_files=[".md"],
+            docker=["clickhouse/stateful-test"],
+        ),
+        run_command='functional_test_check.py "$CHECK_NAME"',
+        runner_type=Runners.FUNC_TESTER,
+        timeout=3600,
```

Comment:
@maxknv Are you sure that 1 hour is enough, especially in case of sanitizers? https://s3.amazonaws.com/clickhouse-test-reports/66439/669ce7cd40881a075fe473027e6668d13683c840/stateful_tests__tsan_.html

```
INFO:root:Going to run func tests: docker run --volume=/home/ubuntu/actions-runner/_work/_temp/test/packages:/package_folder --privileged -e EXTRA_COLUMNS_EXPRESSION="CAST(66439 AS UInt32) AS pull_request_number, '669ce7cd40881a075fe473027e6668d13683c840' AS commit_sha, toDateTime('2024-07-14 08:59:44', 'UTC') AS check_start_time, toLowCardinality('Stateful tests (tsan)') AS check_name, toLowCardinality('m5a.4xlarge') AS instance_type, 'i-02df41d148884f8d8' AS instance_id" -e CLICKHOUSE_CI_LOGS_CREDENTIALS=/tmp/export-logs-config.sh --volume=/home/ubuntu/actions-runner/_work/_temp/test/export-logs-config.sh:/tmp/export-logs-config.sh:ro --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests:/usr/share/clickhouse-test --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/utils/grpc-client:/usr/share/clickhouse-utils/grpc-client --volume=/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/analyzer_tech_debt.txt:/analyzer_tech_debt.txt --volume=/home/ubuntu/actions-runner/_work/_temp/test/result_path:/test_output --volume=/home/ubuntu/actions-runner/_work/_temp/test/server_log:/var/log/clickhouse-server --security-opt seccomp=unconfined --cap-add=SYS_PTRACE -e S3_URL="https://s3.amazonaws.com/clickhouse-datasets" -e AZURE_CONNECTION_STRING='DefaultEndpointsProtocol=https;AccountName=openbucketforpublicci;AccountKey=33RK96U5TjCX9T48wuCu4QukKDuniFfnERcc6v3npAlx1s8oWMKLZCY3u5PKYAzvuVVxoXvvxyqB+AStxV+PUQ==;EndpointSuffix=core.windows.net' -e ADDITIONAL_OPTIONS="--hung-check --print-time" clickhouse/stateful-test:6532a394c436
...
2024-07-14 03:18:32 Using queries from '/usr/share/clickhouse-test/queries' directory
```

So something between this two lines tooks 18 minutes, but usually, this 30mins is enough for this tests - https://play.clickhouse.com/play?user=play#U0VMRUNUIHJlcG9ydF91cmwsIG1pbihjaGVja19zdGFydF90aW1lKSwgbWF4KGNoZWNrX2R1cmF0aW9uX21zKS8xMDAwCkZST00gY2hlY2tzCldIRVJFIHJlcG9ydF91cmwgTElLRSAnJXN0YXRlZnVsX3Rlc3RzX190c2FuXyUnCiAgICBBTkQgY2hlY2tfc3RhcnRfdGltZSA+PSBub3coKSAtIElOVEVSVkFMIDI0IEhPVVIKICAgIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMApHUk9VUCBCWSAxCk9SREVSIEJZIDI=

Reply:
I didn't change timeout values in this PR, I only moved it from one place to another.
We'll take a look into this and align it if needed. Thanks
cc @fm4v